pkg/gardenlet/controller/backupentry/backupentry/reconciler_test.go

```diff
@@ -0,0 +1,253 @@
+		Expect(extensionSecret.Data).To(Equal(gardenSecret.Data))
+
+		Expect(seedClient.Get(ctx, client.ObjectKeyFromObject(extensionBackupEntry), extensionBackupEntry)).To(Succeed())
+		Expect(extensionBackupEntry.Spec).To(MatchFields(IgnoreExtras, Fields{
+			"DefaultSpec": MatchFields(IgnoreExtras, Fields{
+				"Type":           Equal(backupBucket.Spec.Provider.Type),
+				"ProviderConfig": Equal(backupBucket.Spec.ProviderConfig),
+			}),
+			"Region":                     Equal(backupBucket.Spec.Provider.Region),
+			"BackupBucketProviderStatus": Equal(backupBucket.Status.ProviderStatus),
+			"SecretRef": MatchFields(IgnoreExtras, Fields{
+				"Name":      Equal(extensionSecret.Name),
+				"Namespace": Equal(extensionSecret.Namespace),
+			}),
+		}))
+	})
+
+	It("should not reconcile the extension BackupEntry if the secret data or extension spec hasn't changed", func() {
+		Expect(seedClient.Create(ctx, extensionSecret)).To(Succeed())
+		Expect(seedClient.Create(ctx, extensionBackupEntry)).To(Succeed())
+
+		result, err := reconciler.Reconcile(ctx, request)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(result).To(Equal(reconcile.Result{}))
+
+		Expect(seedClient.Get(ctx, client.ObjectKeyFromObject(extensionBackupEntry), extensionBackupEntry)).To(Succeed())
+		Expect(extensionBackupEntry.Annotations).NotTo(HaveKey(v1beta1constants.GardenerOperation))
+	})
+
+	It("should reconcile the extension BackupEntry if the secret data has changed", func() {
+		extensionSecret.Data = map[string][]byte{"dash": []byte("bash")}
+		Expect(seedClient.Create(ctx, extensionSecret)).To(Succeed())
+		Expect(seedClient.Create(ctx, extensionBackupEntry)).To(Succeed())
+
+		result, err := reconciler.Reconcile(ctx, request)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(result).To(Equal(reconcile.Result{}))
+
+		Expect(seedClient.Get(ctx, client.ObjectKeyFromObject(extensionBackupEntry), extensionBackupEntry)).To(Succeed())
+		Expect(extensionBackupEntry.Annotations).To(HaveKeyWithValue(v1beta1constants.GardenerOperation, v1beta1constants.GardenerOperationReconcile))
+	})
+
+	It("should reconcile the extension BackupEntry if the secret update timestamp is after the extension lastupdate time", func() {
+		time := fakeClock.Now().Add(time.Second).UTC().Format(time.RFC3339)
+		metav1.SetMetaDataAnnotation(&extensionSecret.ObjectMeta, v1beta1constants.GardenerTimestamp, time)
+		Expect(seedClient.Create(ctx, extensionSecret)).To(Succeed())
+		Expect(seedClient.Create(ctx, extensionBackupEntry)).To(Succeed())
+
+		result, err := reconciler.Reconcile(ctx, request)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(result).To(Equal(reconcile.Result{}))
+
+		Expect(seedClient.Get(ctx, client.ObjectKeyFromObject(extensionBackupEntry), extensionBackupEntry)).To(Succeed())
+		Expect(extensionBackupEntry.Annotations).To(HaveKeyWithValue(v1beta1constants.GardenerOperation, v1beta1constants.GardenerOperationReconcile))
+	})
```

Comment:
See comment above, please also add a test that considers secrets w/o a timestamp being set.